app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,11 @@
+class LikesController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    unless Like.where(post_id: @post_id, user_id: current_user.id)
+      new_like = current_user.likes.new
+      new_like.post_id = @post.id
+      new_like.save
+    end
```

Comment:
- [ ] I am not able to create a like with this logic as currently implemented. You might want to consider refactoring it like this:
  ```ruby
  unless current_user.likes.find_by(post_id: @post.id)
    @like = current_user.likes.build(post_id: @post.id)
    @like.save
  end
  ```
  
  Please work on this.

Reply:
Ok thank you for your observation will do that.